File: benchmark/lua/polynomial_eval.lua
local start = os.clock()
local coeffs = {}
for i = 1, 1001 do coeffs[i] = i end
local function horner(c, x)
    local result = c[#c]
    for i = #c - 1, 1, -1 do
        result = result * x + c[i]
    end
    return result
end
local s = 0
for _ = 1, 100000 do
    s = s + horner(coeffs, 1.5)
end
local finish = os.clock()
print("Sum: " .. s)
print(string.format("Time: %.0fms", (finish - start) * 1000))

File: benchmark/perl/polynomial_eval.pl
use Time::HiRes qw(time);
my @coeffs = 1..1001;
sub horner {
    my ($coeffs, $x) = @_;
    my $result = $coeffs->[1000];
    for (my $i = 999; $i >= 0; $i--) {
        $result = $result * $x + $coeffs->[$i];
    }
    return $result;
}
my $start = time();
my $sum = 0;
for (1..100000) {
    $sum += horner(\@coeffs, 1.5);
}
my $end = time();
my $ms = ($end - $start) * 1000;
print "Sum: $sum
";
printf "Time: %.0fms
", $ms;
